app/javascript/react/components/ProductFormContainer.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from "react";
+import { Redirect } from "react-router-dom";
+
+const ProductFormContainer = (props) => {
+  const [productRecord, setProductRecord] = useState({
+    name: "",
+    url: "",
+    image_url: "",
+    description: "",
+  });
+
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const handleInputChange = (event) => {
+    setProductRecord({
+      ...productRecord,
+      [event.currentTarget.id]: event.currentTarget.value,
+    });
+  };
+
+  const onSubmitHandeler = (event) => {
+    event.preventDefault();
+    addNewProduct(productRecord);
+  };
+
+  const addNewProduct = (product) => {
+    fetch(`/api/v1/products`, {
+      method: "POST",
+      credentials: "same-origin",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(product),
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status} (${response.statusText})`,
+            error = new Error(errorMessage);
+          throw error;
+        }
+      })
+      .then((response) => response.json())
+      .then((body) => {
+        // debugger;
+      })
+      .catch((error) => console.error(`Error in fetch: ${error.message}`));
+  };
```

Comment:
if you arent doing anything with the returned data until later, than you need remove lines 36 - 50. Otherwise, it looks like you want to redirect to the new page again on 53....but that is redundant since we are already on the page.....so just remove lines 36 - 54, and the redirect state, until you have your index or show page for redirects